presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java

```diff
@@ -608,6 +609,17 @@ public static NullableValue parsePartitionValue(HivePartitionKey key, Type type,
         return parsePartitionValue(key.getName(), key.getValue().orElse(HIVE_DEFAULT_DYNAMIC_PARTITION), type, timeZone);
     }
 
+    public static NullableValue parsePartitionValue(String partitionName, String value, Type type, ZoneId hiveStorageTimeZoneId)
+    {
+        return parsePartitionValue(partitionName, value, type, getDateTimeZone(hiveStorageTimeZoneId));
+    }
+
+    private static DateTimeZone getDateTimeZone(ZoneId hiveStorageTimeZoneId)
+    {
+        requireNonNull(hiveStorageTimeZoneId, "hiveStorageTimeZoneId is null");
```

Comment:
nit: as an convention, we tend to add argument checking code in public method, and omit it in private method.